### ISSUE TITLE
[Feature Request] Add a Low-level API

### ISSUE THREAD

**Issue by AmgadHasan** [mlc-ai/mlc-llm#837]
## 🚀 Feature
It would be really great to expose a low level API that allows us to take full control of the models. Maybe something similar to Huggingface transformers API.
This includes a tokenizer, a model that takes in token_ids and returns logits.

## Motivation

The motivation is this enables people to build complex projects and pipelines using MLC LLM as a backend for text generation

## Alternatives

The current alternative is to dig deep into the code and figure out how to expose the low level LLM

## Additional context

Also it is equally important to have documentation for this low level API with examples (somehing similar to CTranslate2's docs)

**Comment by tqchen** [comment]
https://github.com/mlc-ai/mlc-llm/blob/main/python/mlc_llm/testing/debug_chat.py now contains some low-level api usage. we also enabled a fully openai compatible high level api, closing for now
